packages/myst-frontmatter/src/frontmatter/validators.ts

```diff
@@ -667,21 +667,39 @@ export function validateBinderHubOptions(input: any, opts: ValidationOptions) {
   if (defined(value.url)) {
     output.url = validateUrl(value.url, incrementOptions('url', opts));
   }
-  if (defined(value.ref)) {
-    output.ref = validateString(value.ref, incrementOptions('ref', opts));
-  }
-  if (defined(value.repo)) {
-    output.repo = validateString(value.repo, {
-      ...incrementOptions('repo', opts),
-      regex: GITHUB_USERNAME_REPO_REGEX,
-    });
-  }
   if (defined(value.provider)) {
-    output.provider = validateEnum<BinderProviders>(value.provider, {
+    output.provider = validateString(value.provider, {
       ...incrementOptions('provider', opts),
-      enum: BinderProviders,
+      regex: /.+/,
     });
+    if (output.provider?.match(/^(git|github|gitlab|gist)$/i)) {
+      if (defined(value.repo)) {
+        output.repo = validateString(value.repo, {
+          ...incrementOptions('repo', opts),
+          regex: GITHUB_USERNAME_REPO_REGEX,
+        });
+      }
+    } else {
+      // otherwise provider can be any value, validate as any non empty string
+      output.repo = validateString(value.repo, {
```

Comment:
This doesn't check if `value.repo` is defined. Does that mean it's required in this case (so undefined gives an error?) or should this logic be moved around slightly?

Reply:
yes it is required in this case